test/e2e/framework/gateways.go

```diff
@@ -39,11 +39,14 @@ var gatewayGVR = &schema.GroupVersionResource{
 
 func (f *Framework) AwaitGatewayWithStatus(cluster ClusterIndex, name, status string) *unstructured.Unstructured {
 	gwClient := gatewayClient(cluster)
-	// Set short name without a suffix as exists in the Gateway resource
-	name = strings.Split(name, ".")[0]
 	obj := AwaitUntil(fmt.Sprintf("await Gateway on %q with status %q", name, status),
 		func() (interface{}, error) {
 			resGw, err := gwClient.Get(context.TODO(), name, metav1.GetOptions{})
+			// Some environments sets a node in Gateway resource without a suffix
+			shortName := strings.Split(name, ".")[0]
+			if resGw == nil {
+				resGw, err = gwClient.Get(context.TODO(), shortName, metav1.GetOptions{})
+			}
```

Comment:
It's better to check the return error.  Also to avoid duplication with `AwaitGatewayFullyConnected`, can you put this code in a function `findGateway(cluster ClusterIndex, name string) (*unstructured.Unstructured, error)` ?

```suggestion
			if apierrors.IsNotFound(err) {
			         // Some environments set the Gateway resource name without a suffix
				resGw, err = gwClient.Get(context.TODO(), strings.Split(name, ".")[0], metav1.GetOptions{})
			}
```

Reply:
Done.